### PR TITLE
Document how to use package with Jetpack Glance

### PR DESCRIPTION
**Pull request by tenhobi**
When trying to update my Jetpack Glance from Flutter app, it does not trigger the update. I therefore did update this package so it works with Glance.

Firstly I wanted to make a superclass that would my glance widget had to import, but that did not work. So I changed the code so it uses config where I "register" the Glance widget so the plugin can use the concrete widget class to trigger updates.

This is partially related to #134

//
Btw. while trying out this package, I tested that initial launch as well as widget clicked listener works. 👍 (other things I did not try)

### CODE REVIEW

**Comment by ABausG** [comment]
Thank you so much for this!
Do you know if it is possible to use Glance and the old Provider Way side by side? I'm not sure if that is possible with your current approach? If at all possible in Android..

Would you have time to work on this a bit more? I think we should definitely check that also interactivity is working with Glance.
Also it would be great if you could add some documentation to the readme how to use the features like initialLaunch, Clicked, Interactivity 
If it is possible to use Glance Side by Side with the old way it would be great to also show that in the example

**Comment by tenhobi** [comment]
@ABausG hi, I am not really sure about that. Glance makes a lot of things and even the current widget update does not work. Now I discovered that the update using intent actually works, but still we have to update Preferences to make it work, so Glance probably only refresh widget when the state actually changes? That however does not modify the code since we need to update the Glance state first...

We could probably make custom provider/receiver that extends `GlanceAppWidgetReceiver` and automatically initializes the glance widget (`GlanceAppWidget`) to the config? But that still would not work in cases when you have multiple providers/receivers and want to have different widgets for them...

So I am not really sure how to improve that...

**Comment by tenhobi** [comment]
LMAO I figured it out! It seems using `HomeWidget.updateWidget` the glance receiver's `onUpdate` is called and there I can iterate through all the widgets and update them.

**Comment by tenhobi** [comment]
That's all. This is what I tried/know about Jetpack Glance and this package:

- ❔ saveWidgetData
- ✅ updateWidget
- ❔ getWidgetData
- ⚫ setAppGroupId (iOS)
- ✅ initiallyLaunchedFromHomeWidget
- ✅ widgetClicked
- ❔ registerBackgroundCallback
- ❔ renderFlutterWidget

**Comment by tenhobi** [comment]
@ABausG what do you think about these changes?

**Comment by tenhobi** [comment]
Could you also review this `getIntent` method I need? Since in Jetpack Glance, we can use context.startActivity, which takes Intent, not PendingIntent, I need this method to create just the Intent.

**Comment by Musta-Pollo** [comment]
So you can use glance widgets without a problem?

**Comment by tenhobi** [comment]
Yes, we are using Jetpack Glance for our Flutter app with no problem (we are almost production ready). And we are using this home_widget package to help us with widget updating from an app, handling app initial opening and widget clicks etc. (we are not using that get/save widget data helper since we decided to use shared preferences and secure storage for that directly)

We plan to write (hopefully till the end of the year) an article on [our blog](https://www.netglade.com/en/blog/tag/Flutter), as mentioned in my tweet https://twitter.com/tenhobi/status/1729969425057567149. So I hope that will help people with making a home widget for Flutter apps 👍

**Comment by ABausG** [comment]
Thanks for this contribution @tenhobi here is my plan:

I am preparing right now the release of 0.3.1 which will include some general fixes for Android.
Also probably today or tomorrow I will finish Version 0.4.0 which will make the Interactive iOS stuff transition to a stable release.
My plan would then be to merge the changes from this onto dev channel and finish all this up for Jetpack Compose as Version 0.5.0. My optimistic goal for this would be to get it out in the next week

**Review by ABausG — COMMENTED** [review]
Thank you again so much for this. Apart from the comments some wishes/questions:

1. Could you please make this target the `dev` branch? As I described I plan to release this as 0.5.0 after iOS Interactivity is released (should be beginning of this week) and there are some changes to the README Structure that might affect this PR
2. I think it would be great if we could add an example Widget using Glance to the example App on top of the available Widget. This does not need to be in this PR. Let me know if you would have time for that. Otherwise I should be able to do this as well.
3. Regarding updating the Glance Widget. Have you played around at all if it is somehow possible to detect that whatever class is being found in the updateWidget method in the Plugin and update the state from there? https://developer.android.com/jetpack/compose/glance/glance-app-widget#update-glanceappwidget

File: README.md [hunk 1]
```diff
@@ -58,6 +58,26 @@ which will give you access to the same SharedPreferences
...
+
+If you need to access shared preferences, use this:
+
+```kotlin
+// Shared Preferences used by HomeWidget (this package)
+HomeWidgetPlugin.getData(context)
+
+// Shared Preferences used by Flutter
+context.getSharedPreferences("FlutterSharedPreferences", Context.MODE_PRIVATE)
+```
```
[the remark this answers is not in the document]
Comment (tenhobi) [reply README.md:79]: Hmm, you are probably right! I will delete this note then. :)

**Comment by ABausG** [comment]
@tenhobi Not sure what happened here. But for some reason the PR got closed. Says it is merged however it not really is merged. Would you mind reopening the PR? Think this is mergeable to the dev branch now!

**Comment by tenhobi** [comment]
Interesting. Sure, will do that all on Friday probably

**Comment by ABausG** [comment]
@tenhobi would you mind reopening this (against main)?
I will now work on adding a Glance Widget to the Example App based on the documentation you wrote. That way we should also be able to see which areas could benefit from changes

**Comment by tenhobi** [comment]
I will reopen this from a new branch. I have to recover the changes from the PR.

**Comment by bhoomikshetty** [comment]
> @tenhobi would you mind reopening this (against main)? I will now work on adding a Glance Widget to the Example App based on the documentation you wrote. That way we should also be able to see which areas could benefit from changes

Hello @tenhobi, thankyou so much for this great plugin. Any update on the example app using Glance.

**Comment by ABausG** [comment]
> > @tenhobi would you mind reopening this (against main)? I will now work on adding a Glance Widget to the Example App based on the documentation you wrote. That way we should also be able to see which areas could benefit from changes
> 
> 
> 
> Hello @tenhobi, thankyou so much for this great plugin. Any update on the example app using Glance.
> 
> 

I'm working on it! Hope to get it done soon!!

**Pull request merged**